Feature/4 new type result image

SANHNV opened the pull request:
from existant image in Client, sent in base64 to API and display in UI javascript

HisCodeness committed:
Update README.md

Commit:
Start solution (not working)

Commit:
Start management of Data api

HisCodeness committed:
Make it build

Commit:
Start TypeScript for front js

Commit:
Merge solution

Commit:
Update layout (remove unused scripts)

Commit:
Fix nav node

Commit:
Update Orion TypeScript Configuration
It builds

Commit:
update import and reference types path

Commit:
Migrate to .net Core 3.1

Commit:
Update solution (no big changes)

SANHNV committed:
Continue to rework solution

Commit:
fix post data

Commit:
Beta version

First version beta !!!
It works ;)

SANHNV committed:
Merge from master

SANHNV committed:
image save in Client send in bytes to API and display by UI

SANHNV committed:
uniformized css for Element Detail

SANHNV committed:
Fix useless usings

SANHNV committed:
Fix warning sornarcloud in TakeScreenShotClientScript

SANHNV committed:
Rename byte array Image

mathieumack commented on Orion.Net.Scripts.Common/Diagnostics/TakeScreenShotClientScript.cs:
```diff
@@ -0,0 +1,59 @@
+            var parameter = paramItems.FirstOrDefault(e => e.ParameterName == "screenshot");
+            if (parameter == null)
+            {
+                await SendStringContent("parameter invalid. Screenshot not taken.");
```
Please remove these lines and keep only one call.

mathieumack commented on Orion.Net.Scripts.Common/Diagnostics/TakeScreenShotClientScript.cs:
```diff
@@ -0,0 +1,59 @@
+
+            try
+            {
+                string testPath = @"C:\Users\cadier\Downloads\Logoeconocom.jpg";
```
Please remove all reference to local storage on your laptop

mathieumack replied:
And use the correct file path sended as argument ;)

SANHNV committed:
Change name file, relative path and check parameter

mathieumack commented on Orion.Net.Scripts.Common/Diagnostics/SendImageContentClientScript.cs:
```diff
@@ -10,45 +10,44 @@ namespace Orion.Net.Scripts.Common.Diagnostics
-                await SendImageContent(testPath);
-                await SendStringContent("ScreenShot taken.");
+                await SendImageContent(parameter.ParameterValue);
+                await SendStringContent("Image sent.");
```
Is it needed ?
Please remove it as it's more log information than usefulle information for this script.

mathieumack commented on Orion.Net.Scripts.Common/Diagnostics/SendImageContentClientScript.cs:
```diff
@@ -10,45 +10,44 @@ namespace Orion.Net.Scripts.Common.Diagnostics
-                await SendStringContent("parameter invalid. Screenshot not taken.");
-                await SendStringContent("parameter invalid. Screenshot not taken.");
-                await SendStringContent("parameter invalid. Screenshot not taken.");
+                await SendStringContent("parameter invalid. Image not found.");
```
Please keep only one line. Not 4.

mathieumack commented on Orion.Net.Scripts.Common/Diagnostics/SendImageContentClientScript.cs:
```diff
@@ -10,45 +10,44 @@ namespace Orion.Net.Scripts.Common.Diagnostics
-            var parameter = paramItems.FirstOrDefault(e => e.ParameterName == "screenshot");
-            if (parameter == null)
+            var parameter = paramItems.FirstOrDefault(e => e.ParameterName == "sendImage");
+            if (parameter == null || parameter.ParameterName != "sendImage")
```
To be discussed, we can do it in the BaseClientScript in order to create a generic method and check parameters at each start of Execute method.

SANHNV committed:
Log message deleted

SANHNV committed:
Log message deleted 2

SANHNV committed:
Transfere check parameters in BaseClientScript.
Delete args in SendImageContentClientScript : not use.
Delete SendMessage in OrionHub

mathieumack commented on Orion.Net.Client/Scripts/BaseClientScript.cs:
```diff
@@ -52,6 +52,18 @@ protected async Task SendStringContent(string contentResult)
 
         }
 
+        protected async Task SendImageContent(string pathImage)
```
Could you add a little summary in order to describe the behaviour

mathieumack commented on Orion.Net.Core/Interfaces/ClientScriptResultType.cs:
```diff
@@ -9,6 +9,10 @@ public enum ClientScriptResultType
-        ConsoleLog = 2
+        ConsoleLog = 2,
+        /// <summary>
+        /// byte[], base64
```
byte[] or base64?

mathieumack commented on Orion.Net.Core/Results/ImageContentResult.cs:
```diff
@@ -0,0 +1,22 @@
+        /// <summary>
+        /// console result content
+        /// </summary>
+        public byte[] ImageInByteArray { get; set; }
```
Image**As**ByteArray

SANHNV committed:
Add change and comment.
LoadParam disappear, recoded but error

Commit:
fix async/await call

mathieumack commented on Orion.Net.Client/Configuration/Connector.cs:
```diff
@@ -93,25 +93,27 @@ private BaseClientScript GetCommand(string commandTitle)
             // Send result object to the correct uri :
             var dataUri = string.Empty;
-            HttpContent content = null;
+            HttpContent content = new StringContent(JsonConvert.SerializeObject(result), Encoding.UTF8, "application/json");
```
Please keep the new StringContent in the switch case as some case could generate Something else than a StringContent object.

mathieumack commented on Orion.Net.Client/Scripts/BaseClientScript.cs:
```diff
@@ -20,23 +21,42 @@ public abstract class BaseClientScript : IClientScript
 
         internal async Task Start(string parameters)
         {
-            // Manage start script :
```
Please keep this comment, it prepare a next evolution ;).

mathieumack commented on Orion.Net.Client/Scripts/BaseClientScript.cs:
```diff
@@ -20,23 +21,42 @@ public abstract class BaseClientScript : IClientScript
             // Execute overrided execute method
             await Execute(parameters);
-
-            // Notifiy end script
```
Please keep this comment, it prepare a next evolution ;).

HisCodeness commented on Orion.Net.Scripts.Common/Diagnostics/SendImageContentClientScript.cs:
```diff
@@ -0,0 +1,51 @@
+            identifier = Guid.NewGuid();
+            AvailableParameters.Add(new ScriptParameter()
+            {
+                Name = "filePath"
```
You can create a private constant for this parameter name, this will avoid repeating hard coded string

mathieumack commented on Orion.Net.Client/Scripts/BaseClientScript.cs:
```diff
@@ -47,9 +67,23 @@ protected async Task SendStringContent(string contentResult)
+        /// <param name="pathImage"></param>
+        /// <returns></returns>
+        protected async Task SendImageContent(string pathImage)
+        {
```
Please add some checks about the pathImage
null, StringOrWhitespace or perhaps that the file does not exists

SANHNV committed:
BaseClientScript : add checks for path, file and readablility
Add private const string in ClientScript
